### DISCUSSION TITLE
Is there a way to know tab page is switched in scripts?

### DISCUSSION THREAD

**Discussion by fooyoo** [TriliumNext/Trilium#3348] [Q&A] [answered]
I'd like to capture the event of "tab switch" in script, seems noteSwitched, activeContextChanged is not working for this case.  how to capture some event when I switch the tab? Thanks in advance.

**Comment by zadam** [comment, marked as answer]
[Custom widgets](https://github.com/zadam/trilium/wiki/Custom-widget) can listen to events, the method handlers need to be called with `Event` suffix - e.g.:

```javascript
class WordCountWidget extends api.NoteContextAwareWidget {
     ...

     activeContextChangedEvent() {
    }
}
```

**Reply by fooyoo** [reply]
It works perfectly! Thank you Zadam!
